src/libraries/System.Formats.Tar/tests/TarWriter/TarWriter.WriteEntryAsync.Tests.cs

```diff
@@ -408,5 +408,34 @@ await using (TarWriter writer = new(destination))
 
             AssertExtensions.SequenceEqual(msSource.ToArray(), msDestination.ToArray());
         }
+
+        [Theory]
+        [InlineData(TarEntryFormat.V7)]
+        [InlineData(TarEntryFormat.Ustar)]
+        [InlineData(TarEntryFormat.Gnu)]
+        public async Task WriteEntry_FileSizeOverLegacyLimit_Throws_Async(TarEntryFormat entryFormat)
```

Comment:
This test needs to be split in two separate ones: seekable and unseekable.

For a seekable stream test, you can use a FakeStream that simulates returning a Length > 8Gb. Then when setting an instance to the DataStream property, it should immediately throw if the stream is seekable, length is larger than 8Gb and the entry format is not pax.

For an unseekable stream test, use a MemoryStream, not a FileStream, wrap it in WrappedStream with canSeek=false, and throw.in the WriteEntry method.

Reply:
> Then when setting an instance to the DataStream property, it should immediately throw if the stream is seekable

Throwing in the setter is not meant for this PR, we discussed doing that on https://github.com/dotnet/runtime/pull/75902#discussion_r977921956 and there was not a good reason to do it, we can defer that and apply the same semantic to all properties, not just this one.